Remove script datasource association regression in versions >= 23.1.2

### Description

Since version 23.1.2 there is a regression when attempting to remove a script's datasource association.
After opening the "Select Data Source" window and selecting "None" the script is still listed as being associated to the datasource in the project explorer pane.
Opening it or restarting DBeaver shows that it is indeed no longer associated to the datasource.
Issue was not present in 23.1.1 and is still occuring in the latest 23.2.1 version

### DBeaver Version

Community Edition 23.2.1

### Operating System

_No response_

### Database and driver

_No response_

### Steps to reproduce

1. Start with a script that is associated to a data source
2. Right click and select "Associate with data source"
3. In the "Select Data Source" window use the "None" button
4. Notice that the script still has the original datasource listed in the project explorer "DataSource" column
5. Open the script or restart DBeaver and see that the script is not actually associated to the data source anymore

### Additional context

_No response_

Thanks for the report

```
eclipse.buildId=unknown
java.version=17.0.6
java.vendor=Eclipse Adoptium
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en
Framework arguments:  -eclipse.keyring C:\Users\Dziyana\AppData\Roaming\DBeaverData\secure\secure_storage
Command-line arguments:  -os win32 -ws win32 -arch x86_64

org.eclipse.e4.ui.workbench
Error
Mon Oct 09 11:41:56 CEST 2023
Command 'org.jkiss.dbeaver.core.sql.script.associate' failed

org.eclipse.core.commands.ExecutionException: Error executing 'org.jkiss.dbeaver.core.sql.script.associate': java.lang.NullPointerException: Cannot invoke "org.jkiss.dbeaver.model.DBPDataSourceContainer.getProject()" because "dataSource" is null
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:170)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:488)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:485)
	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:213)
	at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:438)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:449)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:475)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4274)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1066)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4072)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3660)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1155)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:643)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:550)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:171)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:280)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:136)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:402)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:596)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1467)
Caused by: org.eclipse.e4.core.di.InjectionException: java.lang.NullPointerException: Cannot invoke "org.jkiss.dbeaver.model.DBPDataSourceContainer.getProject()" because "dataSource" is null
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:68)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:309)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:243)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:173)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:156)
	... 32 more
Caused by: java.lang.NullPointerException: Cannot invoke "org.jkiss.dbeaver.model.DBPDataSourceContainer.getProject()" because "dataSource" is null
	at org.jkiss.dbeaver.ui.navigator.actions.NavigatorHandlerAssociateScript.execute(NavigatorHandlerAssociateScript.java:77)
	at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:283)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:97)
	at jdk.internal.reflect.GeneratedMethodAccessor108.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	... 36 more


```

--- a/plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/actions/NavigatorHandlerAssociateScript.java
+++ b/plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/actions/NavigatorHandlerAssociateScript.java
@@ -81,7 +81,9 @@ public class NavigatorHandlerAssociateScript extends NavigatorHandlerObjectBase 
                 } else  {
                     project = DBPPlatformDesktop.getInstance().getWorkspace().getProject(script.getProject());
                 }
-                DBNUtils.refreshNavigatorResource(project, script, dataSource);
+                if (project != null) {
+                    DBNUtils.refreshNavigatorResource(project, script, dataSource);
+                }
             }
         }
         return null;

--- a/plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/actions/NavigatorHandlerAssociateScript.java
+++ b/plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/actions/NavigatorHandlerAssociateScript.java
@@ -32,6 +32,7 @@ import org.eclipse.ui.handlers.HandlerUtil;
 import org.eclipse.ui.ide.ResourceUtil;
 import org.jkiss.dbeaver.model.DBPDataSourceContainer;
 import org.jkiss.dbeaver.model.app.DBPPlatformDesktop;
+import org.jkiss.dbeaver.model.app.DBPProject;
 import org.jkiss.dbeaver.model.navigator.DBNNode;
 import org.jkiss.dbeaver.model.navigator.DBNResource;
 import org.jkiss.dbeaver.model.navigator.DBNUtils;
@@ -74,7 +75,13 @@ public class NavigatorHandlerAssociateScript extends NavigatorHandlerObjectBase 
             for (IFile script : selectedScripts) {
                 EditorUtils.setFileDataSource(script, new SimpleDatabaseEditorContext(dataSource));
                 setEditorDataSource(script, dataSource);
-                DBNUtils.refreshNavigatorResource(dataSource.getProject(), script, dataSource);
+                DBPProject project;
+                if (dataSource != null) {
+                    project = dataSource.getProject();
+                } else  {
+                    project = DBPPlatformDesktop.getInstance().getWorkspace().getProject(script.getProject());
+                }
+                DBNUtils.refreshNavigatorResource(project, script, dataSource);
             }
         }
         return null;

